src/common/store/synchronization.slice.ts

```diff
@@ -99,31 +114,53 @@ export const subscribe = (ipc: ReduxIPC) => {
   });
 };
 
-export function* putJson(json: string, isDecrypted: boolean = false): SagaIterator {
+export function* putJson(
+  self: SynchronizationTarget,
+  ipcs: Partial<Record<SynchronizationTarget, ReduxIPC>>,
+  json: string,
+  isDecrypted: boolean = false
+): SagaIterator {
   const [error, action] = safeJSONParse<AnyAction>(json);
   if (error) {
     return;
   }
 
+  console.log(self, 'Received', action);
+
   if (isReduxAction(action) && (isDecrypted || action.type === sendPublicKey.type)) {
     yield put({ ...action, remote: true });
     return;
   }
 
-  const isHandshaken = yield select(getHandshaken);
+  const from = action.from as SynchronizationTarget | undefined;
+  const to = action.to as SynchronizationTarget | undefined;
```

Comment:
What´s the reason for the type cast here?

Reply:
I guess I could type it in the `safeJSONParse`. Would you prefer that?